dotnet/src/Extensions/Planning.ActionPlanner/JsonSerializerContexts.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.Planning.Action;
+
+namespace Microsoft.SemanticKernel.Text;
+
+[JsonSerializable(typeof(ActionPlanResponse))]
+internal sealed partial class SourceGenerationContext : JsonSerializerContext
+{
+    public static readonly SourceGenerationContext WithOptions = new(new JsonSerializerOptions
```

Comment:
My hope is that new code wouldn't accidently use `.Default` when they meant `.WithOptions`. It would be nice to be able to provide _all_ the options for `.Default`.

Reply:
Of the four settings used here, only one is available on the attribute.